### PR TITLE
G-Mode Uses in Red Brinstar: Beta PB, Caterpillar, X-Ray

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

File: region/brinstar/red/Beta Power Bomb Room.json [hunk 1]
```diff
@@ -288,6 +288,35 @@
...
+        }
+      },
+      "requires": [
+        "h_canArtificialMorphPowerBomb"
+      ],
+      "clearsObstacles": ["A", "B"]
+    },
+    {
+      "link": [1, 3],
+      "name": "G-Mode Morph Sidehopper Bomb Kill",
```
Comment (osse101) [inline region/brinstar/red/Beta Power Bomb Room.json:307]: If we just want the runway, we can get most of it by luring the hoppers to the left with Morph.

File: region/brinstar/red/Caterpillar Room.json [hunk 2]
```diff
@@ -1273,6 +1483,46 @@
...
+      "requires": [
+        {"ammo": {"type": "Super", "count": 1}},
+        "Morph"
+      ],
+      "exitCondition": {
+        "leaveWithGMode": {
+          "morphed": false
+        }
+      }
+    },
```
Comment (osse101) [inline region/brinstar/red/Caterpillar Room.json:1504]:
```suggestion
```

Comment (osse101) [reply region/brinstar/red/Caterpillar Room.json:1504]: The gate won't open in GMode

File: region/brinstar/red/X-Ray Scope Room.json [hunk 3]
```diff
@@ -83,6 +83,35 @@
...
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": true
+        }
+      },
+      "requires": [
+        {"or": [
+          "h_canArtificialMorphBombThings",
+          {"and": [
+            "h_canArtificialMorphPowerBomb",
```
Comment (osse101) [inline region/brinstar/red/X-Ray Scope Room.json:99]: The Power Bomb method needs Direct GMode

File: region/brinstar/red/Beta Power Bomb Room.json (hunk 1 above)
Comment (kjbranch) [reply region/brinstar/red/Beta Power Bomb Room.json:307]: That's a good point. For this strat I was mostly thinking if the door was grey - which would wake Zebes

File: region/brinstar/red/X-Ray Scope Room.json [hunk 4]
```diff
@@ -83,6 +83,47 @@
...
+        "After PLMs are overloaded, go through to the item (through the intended entrance or exit path)."
+      ],
+      "devNote": "This is only useful if the item is Morph."
+    },
+    {
+      "link": [1, 2],
+      "name": "G-Mode Morph Overload PLMs (Power Bomb the Item)",
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
```
Comment (osse101) [inline region/brinstar/red/X-Ray Scope Room.json:109]:
```suggestion
          "mode": "direct",
```

**Pull request merged**